collections/src/test/kotlin/ktx/collections/MapsTest.kt

```diff
@@ -393,4 +395,154 @@ class MapsTest {
     assertEquals("Three", gdxArrayMap[3])
     assertEquals("Four", gdxArrayMap[4])
   }
+
+  @Test
+  fun `should return existing value for GdxMap when key exists`() {
+    val map = gdxMapOf("42" to 42)
+
+    val actual = map.getOrPut("42") { 43 }
+
+    assertEquals(42, actual)
+    assertEquals(42, map["42"])
+  }
+
+  @Test
+  fun `should return and put default value to GdxMap when key does not exist`() {
+    val map = gdxMapOf<String, Int>()
+
+    val actual = map.getOrPut("42") { 43 }
+
+    assertEquals(43, actual)
+    assertTrue("42" in map)
+    assertEquals(43, map["42"])
+  }
+
+  @Test
+  fun `should return null for GdxMap when null is stored for given key`() {
+    val map = gdxMapOf<String, Int?>("42" to null)
+
+    val actual = map.getOrPut("42") { 43 }
+
+    assertNull(actual)
+    assertEquals(null, map["42"])
+  }
+
+  @Test(expected = IllegalArgumentException::class)
```

Comment:
In Kotlin we can use `shouldThrow<IllegalArgumentException>` to assert if a certain exception is thrown. In this case it is fine as is, but in general `shouldThrow` is a bit more flexible, as you don't have to end the test with an exception and you can continue with other asserts.

Reply:
See the latest commit.

Reply:
Thank you for the hint. I was looking for assertThrows and something similar and then I found that approach in one of the existing test cases in ktx und just reused it.

Will keep that in mind for the future  :)